sdk/identity/identity/Troubleshooting.md

```diff
@@ -1,47 +1,93 @@
+### AuthenticationRequiredError
 
-## Troubleshoot default Azure credential authentication issues
+Errors arising from authentication issues can be thrown on any service client method that makes a request to the service. This is because the token is requested from the credential on the first call to the service and on any subsequent call that need to refresh the token.
 
-### Credential unavailable
+To distinguish these failures from failures in the service client, Azure Identity classes throws the `AuthenticationRequiredError` with details describing the source of the error in the error message. Depending on the application, these errors may or may not be recoverable.
 
-The `DefaultAzureCredential` attempts to retrieve an access token by sequentially invoking a chain of credentials. In this scenario, the `CredentialUnavailableError` signifies that all credentials in the chain failed to retrieve the token in the current environment setup/configuration. You need to follow the configuration instructions for the respective credential you're looking to use via the `DefaultAzureCredential` chain, so that the credential can work in your environment.
+```ts
+import * from @azure/identity;
+import * from @azure/keyvault-secrets;
+
+// Create a key client using the DefaultAzureCredential
+const keyVaultUrl = `https://key-vault-name.vault.azure.net`;
+const credential = new DefaultAzureCredential();
+const client = new KeyClient(keyVaultUrl, credential);
+
+try {
+// Retrieving the properties of the existing keys in that specific Key Vault.
+  console.log(await client.listPropertiesOfKeys().next());
+} catch (error) {
+  console.log("Authentication Failed", error.message);
+}
+```
+
+### CredentialUnavailableError
+
+The `CredentialUnavailableError` is used to indicate that the credential can't authenticate in the current environment, due to lack of required configuration or setup. This error is also used as a signal to chained credential types, such as `DefaultAzureCredential` and `ChainedTokenCredential`, that the chained credential should continue to try other credential types later in the chain. In `ManagedIdentityCredential` it can also trigger if the authentication endpoints (like the IMDS endpoint) is unavailable.
```

Comment:
```suggestion
The `CredentialUnavailableError` is used to indicate that the credential can't authenticate in the current environment due to lack of required configuration or setup. This error is also used as a signal to chained credential types, such as `DefaultAzureCredential` and `ChainedTokenCredential`, that the chained credential should continue to try other credential types later in the chain. In `ManagedIdentityCredential` it can also trigger if the authentication endpoints (like the IMDS endpoint) is unavailable.
```